tests/test_connection.py

```diff
@@ -49,6 +49,18 @@ def test_connect(self):
         conn = get_connection('testdb')
         self.assertIsInstance(conn, pymongo.mongo_client.MongoClient)
 
+    def test_connect_with_invalid_db_name(self):
+        """Ensure that connect() method fails fast if db name is invalid
+        """
+        with self.assertRaises(InvalidName):
+            connect('mongodb://127.0.0.1:27017')
+
+    def test_connect_with_invalid_db_name_type(self):
+        """Ensure that connect() method fails fast if db name has invalid type
+        """
+        with self.assertRaises(TypeError):
+            connect(['bad type here'])
```

Comment:
Could you also add a test for `'$external'` ? and btw :+1: for thinking about $external

Reply:
added